x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -319,6 +355,10 @@ export class TaskRunner {
 
     const { state, runAt } = await errorAsAlertTaskRunResult(this.loadAlertAttributesAndRun());
 
+    // if subApiKey exists, it should be invalidated after usage
+    if (this.subApiKeyId) {
+      await invalidateAPIKey({ id: this.subApiKeyId }, this.context.securityPluginSetup!);
+    }
```

Comment:
For this, I'm thinking we can move it within `loadAlertAttributesAndRun` somewhere after `validateAndExecuteAlert` has completed. The `apiKey` in that context will be the sub API key that can be invalidated afterwards.

This will also remove the need to store `subApiKeyId` at the class level.